test/web-platform-tests/to-run.yaml

```diff
@@ -753,6 +752,26 @@ template-element/template-content-hierarcy.html: [fail, probably a combination o
 
 ---
 
+DIR: html/semantics/selectors
+
+pseudo-classes/checked-type-change.html: [fail, CSS computed style computation]
+pseudo-classes/dir.html: [fail, nwsapi doesn't support dir]
+pseudo-classes/dir01.html: [fail, nwsapi doesn't support dir]
+pseudo-classes/focus-autofocus.html: [fail, Unknown]
+pseudo-classes/focus.html: [fail, Unknown]
+pseudo-classes/indeterminate-radio.html: [fail, CSS computed style computation]
+pseudo-classes/indeterminate-type-change.html: [fail, CSS computed style computation]
+pseudo-classes/inrange-outofrange-type-change.html: [fail, CSS computed style computation]
+pseudo-classes/inrange-outofrange.html: [fail, Unknown]
+pseudo-classes/link.html: [fail, Unknown]
```

Comment:
This test also [fails the same way](https://wpt.fyi/results/html/semantics/selectors/pseudo-classes/link.html) in browsers. It appears to have relied on elements in `<head>` staying the same despite being shared across tests, and a mass-change inevitably broke it at one point.

Reply:
Nice catch, want to file a bug on web-platform-tests/wpt?

Reply:
Done: https://github.com/web-platform-tests/wpt/issues/11231

Reply:
@domenic @Zirro the other error noticed above in Element-matches.html:

    In-document Element.matches: Universal selector, matching all descendants of the specified reference element (with refNode Element): *

seems to be due to a missing functionality in qS(), qSA() and matches() methods related to the current inability to pass a `refNode` as extra argument that was considered at some point in the past, see here: https://www.w3.org/TR/selectors-api2/#interface-definitions

Current QS API do not include such functionality (passing more params) that I am aware of.

The only current way I know to pass a `refNode` is through the use of the `:scope` pseudo-class so the only fix for current browser would be changing the selector of that test:

 from '*'
 to ':scope *'

I haven't tested it completely yet but:

    document.querySelector('#universal').matches('*'); // return true
    document.querySelector('#universal').matches(':scope *') // return false

Note that this is commented in the file `selectors.js` from Line 415 to Line 445.
More testing and checking are needed, maybe also asking to the test author..

Reply:
Yep, this is a test bug. Being fixed in https://github.com/web-platform-tests/wpt/pull/11229